### DISCUSSION TITLE
problem to get emit client to flask-socketio in docker

### DISCUSSION THREAD

**Discussion by 6273parastu** [miguelgrinberg/Flask-SocketIO#1967] [General]
When i run flask-socketio app in local, It worked fine, but when i deploy project on docker, i do not get emit from client, you think where is the problem?

`Dockerfile : 
`
FROM python:3.9
WORKDIR /socket
COPY . .
RUN python -m pip install --upgrade pip
RUN pip install -r requirements.txt
CMD python run_server.py

`requirements.txt :
`
eventlet==0.30.2
Flask==2.0.3
Flask-Cors==3.0.10
flask-redis==0.4.0
Flask-SocketIO==5.3.3
Flask-WTF==1.1.1
python-engineio==4.3.4
python-socketio==5.7.2

`app.py:
`
```
import eventlet
eventlet.monkey_patch(os=False)
from flask import (Flask, render_template, session, url_for, current_app, request, redirect,
                    flash, jsonify, send_from_directory, send_file)
from flask_socketio import SocketIO
from flask_cors import CORS

app = Flask(__name__)
CORS(app,resources={r"/*":{"origins":"*"}})
socketio = SocketIO()
socketio.init_app(app, async_mode='eventlet', cors_allowed_origins="*")

@socketio.on("connection")
def handle_chat():
    print(request.sid)

if __name__ == '__main__':
    socketio.run(app, port=6310, debug=True, host='0.0.0.0')

**Comment by miguelgrinberg** [comment]
There is nothing in Docker that can cause this. You need to look at the differences between the case that works and the one that doesn't beyond Docker.

**Reply by 6273parastu** [reply]
no difference, I should to say client is react and i problem with connection with react and flask-socketio in docker container.

**Comment by 6273parastu** [comment]
If possible, please write a simple program with flask-socketio and dockerize it and check its connection and emit and send me your project.
the https://www.piesocket.com/socketio-tester is good site for check your connection and emit

**Reply by miguelgrinberg** [reply]
There is an example in this repository that you can use to emit from the client in the browser. You can test with it, if you like.
